### DISCUSSION TITLE
How to set the UID that contains trailing null character.

### DISCUSSION THREAD

**Discussion by shaq23** [fo-dicom/fo-dicom#1686] [Q&A] [answered]
MWL SCP sent Study Instance UID that is padded with a single trailing null character to make even number of bytes in length. When my app set Study Instance UID, then DicomValidation.ValidateUI throws the exception (value contains invalid characters other than '0'-'9' and '.), so my app couldn't set the Study Instance UID value. 
How can my I set the UID that contains a single trailing null ('\0")?
For example, the UID that I received from MWL server is like"1.2.840\0"

**Comment by mrbean-bremen** [comment]
If you set the UID value (without the trailing null), the trailing null will be automatically added during encoding, and the opposite (e.g. the trailing null is stripped) shall happen if you get the value using fo-dicom. 
I guess that you get the value some other way, thereby getting the raw value (including the null) instead.

I'm not sure if it would be a good idea to adapt fo-dicom to accept such a value - there are certainly pros and cons to this.
In the meantime, you can just strip the trailing null yourselves before setting the value.

**Reply by shaq23** [reply]
Thanks for quick response. I also removed the trailing null and set UID, but I'm getting the exception like below. I don't see anything wrong with the value, and there is no leading zeros with "1.2.840.20231121121212.0001".
 -- Exception --
FellowOakDicom.DicomValidationException: Content "1.2.840.20231121121212.0001" does not validate VR UI: components must not have leading zeros
   at FellowOakDicom.DicomValidation.ValidateUI(String content)
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at FellowOakDicom.DicomElement.Validate()
   at FellowOakDicom.DicomDataset.DoAdd(IEnumerable`1 items, Boolean allowUpdate)

My code is like this:
var valueData = "1.2.840.20231121121212.0001";
DicomItem dicomItem = new DicomUniqueIdentifier(DicomTag.StudyInstanceUID, (string[])valueData);
dicomDataset.Add(dicomItem);

**Reply by mrbean-bremen** [reply, marked as answer]
Yes - as the message says: "components must not have leading zeros", but they have in this case. Admittedly, this kind of DICOM standard violation happens often enough, and if you are just copying tags from elsewhere (opposed to creating your owns) it is probably best to switch off validation to avoid this:
```cs
dicomDataset.ValidateItems = false;  // switches off validation for all following added items
var valueData = "1.2.840.20231121121212.0001";
DicomItem dicomItem = new DicomUniqueIdentifier(DicomTag.StudyInstanceUID, (string[])valueData);
dicomDataset.Add(dicomItem);
...
```

**Reply by shaq23** [reply]
My bad. I didn't realize there are leading zeros in the last component. Thanks for your help. :)
